### PR TITLE
#159987647 bookmark article for later

### PR DESCRIPTION
**Pull request by gonuoha**
#### What does this PR do?
add ability to favorite article feature

#### Description of Task to be completed?
- add action, reducer and component for favorite article
- add favorite article button to view article page

#### How should this be manually tested?
Run 'npm install'
Run 'npm run start'
navigate to single article page on the browser and click on options

#### What are the relevant pivotal tracker stories?
[#159987647](https://www.pivotaltracker.com/story/show/159987647)

### CODE REVIEW

**Review by lauragift21 — APPROVED** [review]
LGTM

File: src/actions/favorite.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    type: FAVORITE_LOADING,
+    payload
+})
+
+const success = (payload) => ({
+    type: FAVORITE_SUCCESS,
+    payload
+})
+
+const fialure = (payload) => ({
```
Comment (iamuchejude) [inline src/actions/favorite.js]: Guess you missed the spelling of `failure` here

File: src/components/ArticleContent/ArticleContent.js [hunk 2]
```diff
@@ -32,42 +32,42 @@ class ArticleContent extends Component {
...
+                </span>
               </div>
-              <ArticleTag />
-              <ArticleReaction />
-              <hr className={styles.divider} />
-              <ArticleComment />
+              <div className="my-3 text-secondary">
+                <img src={pointIcon} alt="icon" height="7px" />
+                <span className="pl-1">
+                  {`${(article.timeToRead)} min read`}
```
Comment (iamuchejude) [inline src/components/ArticleContent/ArticleContent.js:57]: What if you have more than 1min, `min` needs to be plural, right?
I will suggest you work that out by checking if `article.timeToRead` is greater than 1 to return `min` or `mins`

**Review by iamuchejude — APPROVED** [review]
Good work

File: src/components/ArticleContent/ArticleContent.js (hunk 2 above)
Comment (AdmiralOnu) [reply src/components/ArticleContent/ArticleContent.js:57]: Thanks for the attention to details Jude. I do think this is the standard, However, you can just slack Gif. she worked on the Article content I believe. I only changed line 65 of this file, not sure why it is showing more changes than that.

Comment (iamuchejude) [reply src/components/ArticleContent/ArticleContent.js:57]: Copied.

File: src/components/Favorite/Favorite.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export class Favorite extends React.Component {
+
+    favorite = () => {
+        const { articleId, hideOptions, dispatch } = this.props
+        hideOptions() // hide more options popup
+        dispatch(addFavorite(articleId))
+    }
+
+    render() {
+        const { setMsg, message } = this.props
```
Comment (johnkingzy) [inline src/components/Favorite/Favorite.js]: why not `setMessage` instead of `setMsg`
```suggestion
        const { setMessage, message } = this.props
```

**Review by johnkingzy — CHANGES_REQUESTED** [review]
👍 Good job.

**Review by johnkingzy — APPROVED** [review]

**Pull request merged**